bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java

```diff
@@ -0,0 +1,242 @@
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.audio.AudioFormat;
+import org.openhab.core.audio.AudioStream;
+import org.openhab.core.audio.ByteArrayAudioStream;
+import org.openhab.core.config.core.ConfigurableService;
+import org.openhab.core.io.net.http.HttpRequestBuilder;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.openhab.core.library.types.RawType;
+import org.openhab.core.voice.TTSException;
+import org.openhab.core.voice.TTSService;
+import org.openhab.core.voice.Voice;
+import org.openhab.voice.mimic.internal.dto.VoiceDto;
+import org.osgi.framework.Constants;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * Mimic Voice service implementation.
+ *
+ * @author Gwendal Roulleau - Initial contribution
+ */
+@Component(configurationPid = MimicTTSService.SERVICE_PID, property = Constants.SERVICE_PID + "="
+        + MimicTTSService.SERVICE_PID)
+@ConfigurableService(category = MimicTTSService.SERVICE_CATEGORY, label = MimicTTSService.SERVICE_NAME
+        + " Text-to-Speech", description_uri = MimicTTSService.SERVICE_CATEGORY + ":" + MimicTTSService.SERVICE_ID)
+@NonNullByDefault
+public class MimicTTSService implements TTSService {
+
+    static final String SERVICE_CATEGORY = "voice";
+    static final String SERVICE_ID = "mimictts";
+    static final String SERVICE_PID = "org.openhab." + SERVICE_CATEGORY + "." + SERVICE_ID;
+    static final String SERVICE_NAME = "Mimic";
+
+    /**
+     * Configuration parameters
+     */
+    private static final String PARAM_URL = "url";
+    private static final String PARAM_SPEAKINGRATE = "speakingRate";
+    private static final String PARAM_AUDIOVOLATITLITY = "audioVolatility";
+    private static final String PARAM_PHONEMEVOLATITLITY = "phonemeVolatility";
+
+    /**
+     * Url
+     */
+    private static final String LIST_VOICES_URL = "/api/voices";
+    private static final String SYNTHETIZE_URL = "/api/tts";
+
+    /** The only wave format supported */
+    private static final AudioFormat AUDIO_FORMAT = new AudioFormat(AudioFormat.CONTAINER_WAVE,
+            AudioFormat.CODEC_PCM_SIGNED, false, 16, 52000, 22050L, 1);;
+
+    /**
+     * Logger.
+     */
+    private final Logger logger = LoggerFactory.getLogger(MimicTTSService.class);
+
+    private final MimicConfiguration config;
+
+    private final Gson gson = new GsonBuilder().create();
+
+    @Activate
+    public MimicTTSService(final @Reference ConfigurationAdmin configAdmin) {
```

Comment:
You do not need to reference ConfigurationAdmin and do what you do in the body of the method.
Method `activate` will be called with a map. Look at Polly TTS or VoiceRSS as examples.

In fact, this method can be removed if you initialize `config` in its declaration.
```
private final MimicConfiguration config = new MimicConfiguration();
```

Reply:
Good to know, thanks. Done and tested OK.